src/ui/profile/BeyondLoader.js

```diff
@@ -0,0 +1,118 @@
+import React, { useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { parseBeyondStatus, parseBeyondSheet } from '../../utils/parseBeyondSheet';
+
+const BEYOND_MAX_RETRIES = 60;
+
+function BeyondLoader(props) {
+  const { currentProfile, onBeyondLoaded } = props;
+  if (!currentProfile) {
+    return null;
+  }
+
+  const url = `https://www.dndbeyond.com/characters/${currentProfile.id}`;
+  return (
+    <BeyondFrame
+      url={url}
+      onBeyondLoaded={onBeyondLoaded}
+    />
+  );
+}
+
+BeyondLoader.propTypes = {
+  onBeyondLoaded: PropTypes.func.isRequired,
+  currentProfile: PropTypes.shape({
+    avatar: PropTypes.string.isRequired,
+    id: PropTypes.number.isRequired,
+    level: PropTypes.number.isRequired,
+    name: PropTypes.string.isRequired,
+  }),
+};
+
+BeyondLoader.defaultProps = {
+  currentProfile: {
+    avatar: '',
+    id: null,
+    level: null,
+    name: '',
+  },
+};
+
+// There are two instances of loading:
+// 1) The `iframe` fetches the `dndbeyond.com/characters/id` page
+// 2) The page dynamically loads character sheet data
+// We need to ensure that step 2 is finished before attempting to parse
+async function checkSheetLoaded(frameDocument) {
```

Comment:
I know you told me to ignore this file, but here's some advice anyway:

This function does not need to be `async` because it doesn't use `await` in its body. This function is effectively the same as the following:

```js
function checkSheetLoaded(frameDocument) {
    ... same body ...
    } else {
        return Promise.reject(new Error('RETRY'));
    }
    return Promise.resolve(sheetResult);
}
```